sympy/logic/boolalg.py

```diff
@@ -599,6 +621,20 @@ def _eval_as_set(self):
         from sympy.sets.sets import Union
         return Union(*[arg.as_set() for arg in self.args])
 
+    def to_anf(self, deep=True):
+        n_args = len(self.args)
+        a = self.args[:int(n_args/2)]
+        b = self.args[int(n_args/2):]
+        if len(a) == 1:
+            a = a[0]
+        else:
+            a = to_anf(Or(*a), deep=deep)
+        if len(b) == 1:
+            b = b[0]
+        else:
+            b = to_anf(Or(*b), deep=deep)
```

Comment:
for this as well is missing.

Reply:
This was refactored as

    def to_anf(self, deep=True):
        args = range(1, len(self.args) + 1)
        args = (combinations(self.args, j) for j in args)
        args = chain.from_iterable(args)  # powerset
        args = (And(*arg) for arg in args)
        args = map(lambda x: to_anf(x, deep=deep) if deep else x, args)
        return Xor(*list(args), remove_true=False)

And covered with tests

    assert to_anf(Or(x, y)) == Xor(x, y, And(x, y))
    assert to_anf(Or(Implies(x, y), And(x, y), y)) == \
            Xor(x, True, x & y, remove_true=False)
    assert to_anf(Or(Nand(x, y), Nor(x, y), Xnor(x, y), Implies(x, y))) == True
    assert to_anf(Or(x, Not(y), Nor(x,z), And(x, y), Nand(y, z))) == \
            Xor(True, And(y, z), And(x, y, z), remove_true=False)